cdk/cdk/cdk_stack.py

```diff
@@ -237,8 +123,136 @@ def __init__(self, scope: core.Construct, construct_id: str, **kwargs) -> None:
+            code=aws_lambda.Code.from_asset_image(
+                directory="lambdas/ndjson-builder",
+                file="Dockerfile",
+                entrypoint=["/usr/local/bin/python", "-m", "awslambdaric"],
+                cmd=["handler.handler"],
+            ),
+            handler=aws_lambda.Handler.FROM_IMAGE,
+            runtime=aws_lambda.Runtime.FROM_IMAGE,
+            memory_size=8000,
+            timeout=core.Duration.minutes(10),
+            environment={
+                "BUCKET": ndjson_bucket.bucket_name,
+                "QUEUE_URL": ndjson_queue.queue_url,
+                "COLLECTION": "HLSS30.002"
+            },
+        )
+
+        item_event_source = SqsEventSource(
+            ingest_queue,
+            batch_size=100,
+            max_batching_window=core.Duration.seconds(300),
+            report_batch_item_failures=True,
+        )
+        build_ndjson_function.add_event_source(item_event_source)
+
+        pgstac_loader_role = aws_iam.Role(
+            self,
+            "PGStacLoaderRole",
+            assumed_by=aws_iam.ServicePrincipal("lambda.amazonaws.com"),
+            managed_policies=[
+                aws_iam.ManagedPolicy.from_aws_managed_policy_name(
+                    "service-role/AWSLambdaBasicExecutionRole"
+                )
+            ],
+        )
+
+        ndjson_queue.grant_consume_messages(pgstac_loader_role)
+
+        ndjson_bucket.grant_read(pgstac_loader_role)
+
+        pgstac_secret = secretsmanager.Secret.from_secret_arn(
+            self,
+            id="PGStacSecret",
+            secret_arn=SECRET_NAME,
+        )
+
+        pgstac_secret.grant_read(pgstac_loader_role)
+
+        pgstac_security_group = ec2.SecurityGroup(
+            self,
+            f"{id}-pgstac-loader-sg",
+            vpc=database_vpc,
+            description="fromCogPipelinesPgstacLoader",
+        )
+
+
+        pgstac_loader = aws_lambda.Function(
+            self,
+            f"{id}-pgstac-loader-lambda",
+            role=pgstac_loader_role,
+            code=aws_lambda.Code.from_asset_image(
+                directory="lambdas/pgstac-loader",
+                file="Dockerfile",
+                entrypoint=["/usr/local/bin/python", "-m", "awslambdaric"],
+                cmd=["handler.handler"],
+            ),
+            handler=aws_lambda.Handler.FROM_IMAGE,
+            runtime=aws_lambda.Runtime.FROM_IMAGE,
+            memory_size=8000,
+            timeout=core.Duration.minutes(10),
+            environment={
+                "SECRET_NAME": SECRET_NAME,
+            },
+            reserved_concurrent_executions=3,
+            vpc=database_vpc,
+            vpc_subnets=ec2.SubnetSelection(subnet_type=ec2.SubnetType.PRIVATE),
+            security_groups=[lambda_function_security_group],
+
+        )
+        pgstac_loader.add_to_role_policy(ec2_network_access)
+
+        ndjson_event_source = SqsEventSource(
+            ndjson_queue,
+            batch_size=5,
```

Comment:
Since this code will soon be specific to our HLS pipeline, I think this is ok but if we want a more generic pipeline I think something like `batch_size` and other items such as `max_batching_window` will need to be configurable.